src/plugins/workflow_designer/src/WorkflowViewController.cpp

```diff
@@ -1974,9 +1995,10 @@ void WorkflowView::sl_saveScene() {
         sl_updateTitle();
     }
     propertyEditor->commit();
-    Task *t = new SaveWorkflowSceneTask(getSchema(), getMeta());
-    AppContext::getTaskScheduler()->registerTopLevelTask(t);
-    connect(t, SIGNAL(si_stateChanged()), SLOT(sl_onSceneSaved()));
+    HRSchemaSerializer::updateWorkflowSchemaPathSettings(meta);
+    U2OpStatus2Log os;
+    HRSchemaSerializer::saveSchema(schema.get(), &meta, meta.url, os);
```

Comment:
Please check the state of the 'os' separately from scene->setModified().
The scene->isModified should not change (to either false or true) if the saveSchema is failed.